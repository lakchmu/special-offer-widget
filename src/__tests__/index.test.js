/* global jest, it, expect */

import TBFSpecialOffer from '../index';

jest.mock('../TBFSpecialOffers/request');
jest.mock('../index.css');

function setup() {
  return {
    targetDivElement: '.special-offers',
    token: 'KO96pMbbIBTQPl7vnoJMXmIrEl2cYoUZS83QlL0Abu0',
  };
}

it('Test TBFSpecialOffer function with default settings', () => {
  document.body.innerHTML = '<div class="special-offers"></div>';
  const setupData = setup();
  expect.assertions(1);
  return TBFSpecialOffer(setupData.targetDivElement, setupData.token)
    .then(offersView => expect(offersView).toMatchSnapshot());
});
