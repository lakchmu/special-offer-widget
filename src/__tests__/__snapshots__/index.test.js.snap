// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test TBFSpecialOffer function with default settings 1`] = `
OffersView {
  "options": Object {
    "defaultCSS": true,
    "defaultEvents": true,
  },
  "rootElement": <div
    class="special-offers tbf-so-special-offers"
  >
    
  
    <div
      class="tbf-so-offer"
    >
      
    
      <div
        class="tbf-so-offer__content"
      >
        
      
        <div
          class="tbf-so-offer__content-image"
        >
          
        
          <img
            class="tbf-so-offer__image "
            src="https://s3.amazonaws.com/buuqit-dev/special_offers/images/000/000/009/public/Room2.jpg?1515606281"
          />
          
      
        </div>
        
      
        <div
          class="tbf-so-offer__content-text"
        >
          
        
          <div
            class="tbf-so-offer__title"
          >
            Last Minute Booking
          </div>
          
        
          <div
            class="tbf-so-offer__wrapper-description"
          >
            
          
            <div
              class="tbf-so-offer__short-description"
            >
              Last minute rate, Non-Refundable
            </div>
            
          
            <div
              class="tbf-so-offer__description"
            >
              Last minute booking discount, full pre payment required on time of booking. Please note this rate is non-refundable.
            </div>
            
        
          </div>
          
        
          <div
            class="tbf-so-offer__footer"
          >
            
          
            <span
              class="tbf-so-offer__dates"
            >
              
            
              <i
                class="tbf-so-offer__icon-calendar"
              />
              Jun 9 - Aug 31
          
            </span>
            
          
            <span
              class="tbf-so-offer__control-links"
            >
              
            
              <button
                class="tbf-so-offer__more-link"
              >
                More
              </button>
              
            
              <a
                class="tbf-so-offer__booking-link"
                href="https://app.thebookingfactory.com/Spilman-Hotel/book/__special_offer_9"
              >
                Book now
              </a>
              
          
            </span>
            
        
          </div>
          
      
        </div>
        
    
      </div>
      
  
    </div>
    


  
    <div
      class="tbf-so-offer"
    >
      
    
      <div
        class="tbf-so-offer__content"
      >
        
      
        <div
          class="tbf-so-offer__content-image"
        >
          
        
          <img
            class="tbf-so-offer__image "
            src="https://s3.amazonaws.com/buuqit-dev/special_offers/images/000/000/006/public/Room7.jpg?1512864349"
          />
          
      
        </div>
        
      
        <div
          class="tbf-so-offer__content-text"
        >
          
        
          <div
            class="tbf-so-offer__title"
          >
            Stay 3 Nights Save 15%
          </div>
          
        
          <div
            class="tbf-so-offer__wrapper-description"
          >
            
          
            <div
              class="tbf-so-offer__short-description"
            >
              Stay 3 Nights Save 15%
            </div>
            
          
            <div
              class="tbf-so-offer__description"
            >
              Stay for 3 nights or more and save 15%. Save a little by staying with us a bit longer
            </div>
            
        
          </div>
          
        
          <div
            class="tbf-so-offer__footer"
          >
            
          
            <span
              class="tbf-so-offer__dates"
            >
              
            
              <i
                class="tbf-so-offer__icon-calendar"
              />
              Jun 4 - Aug 31
          
            </span>
            
          
            <span
              class="tbf-so-offer__control-links"
            >
              
            
              <button
                class="tbf-so-offer__more-link"
              >
                More
              </button>
              
            
              <a
                class="tbf-so-offer__booking-link"
                href="https://app.thebookingfactory.com/Spilman-Hotel/book/__special_offer_6"
              >
                Book now
              </a>
              
          
            </span>
            
        
          </div>
          
      
        </div>
        
    
      </div>
      
  
    </div>
    

  </div>,
  "storage": OffersStorage {
    "list": Array [
      OfferModel {
        "bookingLink": "https://app.thebookingfactory.com/Spilman-Hotel/book/__special_offer_9",
        "dateFrom": 2016-06-09T00:00:00.000Z,
        "dateTo": 2018-08-31T00:00:00.000Z,
        "description": "Last minute booking discount, full pre payment required on time of booking. Please note this rate is non-refundable.",
        "discountType": "percent",
        "discountValue": "15.0",
        "imageLink": "https://s3.amazonaws.com/buuqit-dev/special_offers/images/000/000/009/public/Room2.jpg?1515606281",
        "shortDescription": "Last minute rate, Non-Refundable",
        "title": "Last Minute Booking",
      },
      OfferModel {
        "bookingLink": "https://app.thebookingfactory.com/Spilman-Hotel/book/__special_offer_6",
        "dateFrom": 2016-06-04T00:00:00.000Z,
        "dateTo": 2018-08-31T00:00:00.000Z,
        "description": "Stay for 3 nights or more and save 15%. Save a little by staying with us a bit longer",
        "discountType": "percent",
        "discountValue": "15.0",
        "imageLink": "https://s3.amazonaws.com/buuqit-dev/special_offers/images/000/000/006/public/Room7.jpg?1512864349",
        "shortDescription": "Stay 3 Nights Save 15%",
        "title": "Stay 3 Nights Save 15%",
      },
    ],
  },
}
`;
